rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+
+## a) Hash functions in Nescience 
+Hash functions are a foundational element of Nescience’s cryptographic framework, serving multiple critical roles that ensure the security, privacy, and efficiency of the system. 
+One of the primary uses of hash functions in Nescience is to conceal sensitive details of UTXOs by converting them into fixed-size hashes. This process allows UTXO details 
+to remain private, ensuring that sensitive information is not directly exposed on the blockchain, while still enabling their existence and integrity to be verified. Hashing 
+the UTXO details allows the actual data to remain confidential, with the hashes stored in a global tree structure for efficient management and retrieval.
+
+Additionally, hash functions are essential for generating **nullifiers**, which play a crucial role in preventing double-spending. Nullifiers are created by hashing UTXOs 
+and are used to mark them as spent, ensuring that they cannot be reused in subsequent transactions. These nullifiers are stored in a nullifier tree, and each transaction 
+must prove that its UTXO’s nullifier is not already present in the tree before it can be processed. This ensures that the UTXO has not been spent before, maintaining the 
+integrity of the transaction process.
+
+Hash functions are also vital in the construction of **sparse Merkle trees**, which provide an efficient and secure method for verifying data within the blockchain. 
+Sparse Merkle trees enable quick and reliable proofs of membership and non-membership, making them essential for verifying both UTXOs and nullifiers. By using hash functions 
+to build these trees, Nescience can ensure the integrity of the data, as any tampering with the data would result in a change in the hash, making the manipulation detectable.
+
+Another critical consideration in Nescience is the compatibility of hash functions with **zero-knowledge proofs**. ZK-friendly hash functions are optimized for efficient 
+computation within the constraints of ZK circuits, ensuring that they do not become a bottleneck in the proof generation or verification process. These hash functions 
+maintain strong cryptographic security properties while enabling efficient computations in zero-knowledge proof systems, which is essential for maintaining privacy and 
+integrity within the ZK framework.
+
+The primary advantage of using hash functions in Nescience is their ability to ensure that transaction details remain private while still allowing for verification 
+of their validity. Furthermore, by integrating hash functions into Merkle trees, the blockchain data becomes tamper-proof, enabling quick and efficient verification 
+processes that uphold the system’s security and privacy standards.
+
+### Use case: How to use the Pedersen hash to create the UTXO commitment
+---
+As mentioned in the [UTXOs in Private Exections section](#pe), the user broadcasts the encrypted UTXOs to the network, along with a commitment to the output UTXOs 
```

Comment:
Capitalisation here:
"UTXOs in Private Exections section" (should be lowercase). Also, typo on "executions"